state/store_test.go

```diff
@@ -329,24 +333,16 @@ func TestSaveRetainHeight(t *testing.T) {
 }
 
 func TestMinRetainHeight(t *testing.T) {
-	stateDB := dbm.NewMemDB()
-	stateStore := sm.NewStore(stateDB, sm.StoreOptions{
-		DiscardABCIResponses: false,
-	})
-	require.NoError(t, initStateStoreRetainHeights(stateStore, 0, 0, 0))
-	pruner := sm.NewPruner(
-		stateStore,
-		nil,
-		log.TestingLogger(),
-		sm.WithPrunerCompanionEnabled(),
-	)
+	_, bs, txIndexer, blockIndexer, callbackF, stateStore := makeStateAndBlockStoreAndIndexers()
+	defer callbackF()
+	pruner := sm.NewPruner(stateStore, bs, blockIndexer, txIndexer, log.TestingLogger())
 	minHeight := pruner.FindMinRetainHeight()
 	require.Equal(t, int64(0), minHeight)
 
 	err := stateStore.SaveApplicationRetainHeight(10)
 	require.NoError(t, err)
 	minHeight = pruner.FindMinRetainHeight()
-	require.Equal(t, int64(0), minHeight)
+	require.Equal(t, int64(10), minHeight)
```

Comment:
This should not be changed? How is the test passing after this change? By default the data companion retain height is set o nstartup to 0, which means no pruning will occur and it should not return 10 as the minimum retain height.